lib/commons/text/unicode.js

```diff
@@ -130,6 +133,7 @@ function getSupplementaryPrivateUseRegExp() {
 	/**
 	 * Reference: https://www.unicode.org/charts/PDF/UD800.pdf
 	 * https://www.unicode.org/charts/PDF/UDC00.pdf
+	 * https://www.unicode.org/charts/PDF/UF0000.pdf
 	 */
-	return /[\uDB80-\uDBBF][\uDC00-\uDFFD]/g;
+	return /[\uDB80-\uDBBF][\uDC00-\uDFFD]|[\u{F0000}-\u{FFFFD}]/gu;
```

Comment:
So the `u` flag might pose a problem in some situations in IE11 as it's [not supported](https://kangax.github.io/compat-table/es6/#test-RegExp_y_and_u_flags). We've had quite the issue with these regexs before in IE11, so probably best to change this to use the surrogate pair codes as the `[\uDB80-\uDBBF][\uDC00-\uDFFD]` is doing.

Reply:
Babel's preset-env is compiling this to a regex without the `u` flag, and we should be good with regards to IE11 support.

The compiled axe.js has:

```js
function getSupplementaryPrivateUseRegExp() {
    return /(?:[\uDB80-\uDBBF](?![\uDC00-\uDFFF]))(?:(?:[^\uD800-\uDBFF]|^)[\uDC00-\uDFFD])|(?:[\uDB80-\uDBBE][\uDC00-\uDFFF]|\uDBBF[\uDC00-\uDFFD])/g;
}
```

That's obviously a little bit gnarlier than it was before (especially with the added non-capturing groups).

Manually writing the surrogate pair I think would give us something like:

```js
function getSupplementaryPrivateUseRegExp() {
    return /[\uDB80-\uDBBF][\uDC00-\uDFFD]|(?:[\uDB80-\uDBBE][\uDC00-\uDFFF]|\uDBBF[\uDC00-\uDFFD])/g;
}
```

Because the regex gets compiled to a regex compatible with ES5, it may be preferable to keep the unicode escape sequences in the regex, since it's easier to see what the actual range is.

Do the windows tests run in IE11, or on Edge?

Or, if you still prefer manually writing the surrogate pair, I'll do it. Let me know! Thanks

Reply:
Actually, I'll just update it to

```js
/[\uDB80-\uDBBF][\uDC00-\uDFFD]|[\udb80-\udbbe][\udc00-\udfff]|\udbbf[\udc00-\udffe]]/g
```

to be safe. The `u` flag does get compiled to a compatible regex, but with the surrogate pairs there's no ambiguity as to what's happening.